src/battle/anim/ghost.c

```diff
@@ -189,3 +202,402 @@ const struct SpriteTemplate gSpriteTemplate_83DAF98 =
+        gTasks[taskId].data[10] = 0;
+        gTasks[taskId].data[2] += 1;
+        gTasks[taskId].data[3] -= 1;
+        REG_BLDALPHA = gTasks[taskId].data[3] << 8 | gTasks[taskId].data[2];
+        if (gTasks[taskId].data[2] != 9)
+            return;
+        gTasks[taskId].func = &sub_80DDED0;
+    }
+}
+
+void sub_80DDED0(u8 taskId) {
+    u8 r1;
+    if (gTasks[taskId].data[1] > 0) {
+        gTasks[taskId].data[1] -= 1;
+        return;
+    }
+    r1 = GetAnimBattlerSpriteId(0);
+    gTasks[taskId].data[0] += 8;
+    if (gTasks[taskId].data[0] <= 0xFF) {
+        obj_id_set_rotscale(r1, gTasks[taskId].data[0], gTasks[taskId].data[0], 0);
+        return;
+    }
+    sub_8078F40(r1);
+    DestroyAnimVisualTask(taskId);
+    REG_BLDCNT = 0;
+    REG_BLDALPHA = 0;
+}
+
+void sub_80DDF40(struct Sprite *sprite) {
+    u16 r5, r6;
+    r5 = sprite->pos1.x;
+    r6 = sprite->pos1.y;
+    sprite->pos1.x = GetBattlerSpriteCoord(gAnimBankAttacker, 2);
+    sprite->pos1.y = GetBattlerSpriteCoord(gAnimBankAttacker, 3);
+    sprite->data[0] = 0;
+    sprite->data[1] = gBattleAnimArgs[0];
+    sprite->data[2] = gBattleAnimArgs[1];
+    sprite->data[3] = gBattleAnimArgs[2];
+    sprite->data[4] = sprite->pos1.x << 4;
+    sprite->data[5] = sprite->pos1.y << 4;
+    sprite->data[6] = (((s16)r5 - sprite->pos1.x) << 4) / (gBattleAnimArgs[0] << 1);
+    sprite->data[7] = (((s16)r6 - sprite->pos1.y) << 4) / (gBattleAnimArgs[0] << 1);
+    sprite->callback = &sub_80DDFE8;
+}
+
+void sub_80DDFE8(struct Sprite *sprite) {
+    switch (sprite->data[0]) {
+    case 0:
+        sprite->data[4] += sprite->data[6];
+        sprite->data[5] += sprite->data[7];
+        sprite->pos1.x = sprite->data[4] >> 4;
+        sprite->pos1.y = sprite->data[5] >> 4;
+        sprite->data[1] -= 1;
+        if (sprite->data[1] > 0)
+            break;
+        sprite->data[0] += 1;
+        break;
+    case 1:
+        sprite->data[2] -= 1;
+        if (sprite->data[2] > 0)
+            break;
+        sprite->data[1] = GetBattlerSpriteCoord(gAnimBankTarget, 2);
+        sprite->data[2] = GetBattlerSpriteCoord(gAnimBankTarget, 3);
+        sprite->data[4] = sprite->pos1.x << 4;
+        sprite->data[5] = sprite->pos1.y << 4;
+        sprite->data[6] = ((sprite->data[1] - sprite->pos1.x) << 4) / sprite->data[3];
+        sprite->data[7] = ((sprite->data[2] - sprite->pos1.y) << 4) / sprite->data[3];
+        sprite->data[0] += 1;
+        break;
+    case 2:
+        sprite->data[4] += sprite->data[6];
+        sprite->data[5] += sprite->data[7];
+        sprite->pos1.x = sprite->data[4] >> 4;
+        sprite->pos1.y = sprite->data[5] >> 4;
+        sprite->data[3] -= 1;
+        if (sprite->data[3] > 0)
+            break;
+        sprite->pos1.x = GetBattlerSpriteCoord(gAnimBankTarget, 2);
+        sprite->pos1.y = GetBattlerSpriteCoord(gAnimBankTarget, 3);
+        sprite->data[0] += 1;
+        break;
+    case 3:
+        move_anim_8074EE0(sprite);
```

Comment:
`break` should be present, unless it's explicitly not supposed to be there based on the program flow.

Reply:
At the end of `case 3`? Isn't that redundant?

Reply:
Not in my opinion.  What if I come here and add a case 4, but don't realize case 3 wasn't breaking.